### PR TITLE
libazureinit: Add README

### PR DESCRIPTION
**Pull request by dongsupark**
Add initial README in `libazureinit`, to be published to its `crates.io` website as well.

Add a link to `libazureinit` on the top-level `azure-init` README.

### CODE REVIEW

File: libazureinit/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# libazureinit
+
+A common library for provisioning Linux VMs on Azure.
+
+Features would include like:
```
Comment (anhvoms) [inline libazureinit/README.md:5]:
```suggestion
Features:
```

File: libazureinit/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+* minimal configurations of virtual machines
```
Comment (anhvoms) [inline libazureinit/README.md:7]:
```suggestion
* retrieve provisioning metadata from Azure Instance Metadata Service
* configure the VM according to the provisioning metadata
* report provisioning complete to Azure platform
```

File: libazureinit/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+* provisioning metadata
+* basic features for instance initialisation
+
+On the other hand, [azure-init](https://github.com/Azure/azure-init) is a reference implementation which actually calls API provided by libazureinit.
```
Comment (anhvoms) [inline libazureinit/README.md:11]:
```suggestion
[azure-init](https://github.com/Azure/azure-init) is reference implementation that leverages the APIs provided by libazureinit.
```

File: libazureinit/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+Goal is to make the library provide common API to be used by other provisioning implementations.
```
Comment (anhvoms) [inline libazureinit/README.md:13]: The goal is to provide APIs to for other components to perform VM provisioning on Azure platform.

**Review by anhvoms — CHANGES_REQUESTED** [review]

**Comment by dongsupark** [comment]
Done, thanks.

File: libazureinit/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+A common library for provisioning Linux VMs on Azure.
+
+Features:
+
+* retrieve provisioning metadata from Azure Instance Metadata Service
+* configure the VM according to the provisioning metadata
+* report provisioning complete to Azure platform
+* basic features for instance initialisation
+
+[azure-init](https://github.com/Azure/azure-init) is reference implementation that leverages the APIs provided by libazureinit.
```
Comment (anhvoms) [inline libazureinit/README.md:12]:
```suggestion
[azure-init](https://github.com/Azure/azure-init) is a reference implementation that leverages the APIs provided by libazureinit.
```

**Review by anhvoms — APPROVED** [review]

**Pull request merged**